rust-semantics/expression/loops.md

```diff
@@ -0,0 +1,50 @@
+```k
+
+module RUST-LOOP-EXPRESSIONS
+    imports RUST-REPRESENTATION
+
+    syntax IteratorLoopExpression ::=  "for1" Pattern "limit" PtrValue BlockExpression
+                                    | "for2" Pattern "limit" PtrValue BlockExpression
+
+    rule for Patt:Identifier:PatternNoTopAlt | R:PatternNoTopAlts in ptrValue(_, intRange(First, Last)) B:BlockExpression => 
+            {
+                .InnerAttributes
+                let Patt = ptrValue(null, First);
+                (for1 Patt | R limit ptrValue(null, Last) B):IteratorLoopExpression; 
+                .NonEmptyStatements
+            };
+        requires checkIntOfSameType(First, Last)
+
+    rule for1 Patt:Identifier:PatternNoTopAlt | .PatternNoTopAlts limit Last B:BlockExpression => 
+                if (Patt :: .PathExprSegments):PathExprSegments < Last { .InnerAttributes B; (for2 Patt:Identifier:PatternNoTopAlt | .PatternNoTopAlts limit Last B):IteratorLoopExpression;  .NonEmptyStatements};
```

Comment:
I would split this into multiple lines.

Also, you can probably inline the "for2" rule below if you want.